sdk/core/Azure.Core/tests/BearerTokenAuthenticationPolicyTests.cs

```diff
@@ -120,30 +75,264 @@ public async Task CachesHeaderValue()
         }
 
         [Test]
-        public void ThrowsForNonTlsEndpoint()
+        public void BearerTokenAuthenticationPolicy_ThrowsForNonTlsEndpoint()
         {
-            var credentialsMock = new Mock<TokenCredential>();
-            if (IsAsync)
+            var credential = new TokenCredentialStub(
+                (r, c) => r.Scopes.SequenceEqual(new[] { "scope" }) ? new AccessToken("token", DateTimeOffset.MaxValue) : default, IsAsync);
+
+            var policy = new BearerTokenAuthenticationPolicy(credential, "scope");
+            MockTransport transport = CreateMockTransport();
+
+            Assert.ThrowsAsync<InvalidOperationException>(async () => await SendGetRequest(transport, policy, uri: new Uri("http://example.com")));
+        }
+
+        [Test]
+        public async Task BearerTokenAuthenticationPolicy_OneHundredConcurrentCalls()
+        {
+            var credential = new TokenCredentialStub((r, c) =>
             {
-                credentialsMock.Setup(
-                        credential => credential.GetTokenAsync(
-                            It.Is<TokenRequestContext>(request => request.Scopes.SequenceEqual(new[] { "scope" })),
-                            It.IsAny<CancellationToken>()))
-                    .ReturnsAsync(new AccessToken("token", DateTimeOffset.MaxValue));
+                Thread.Sleep(100);
+                return new AccessToken(Guid.NewGuid().ToString(), DateTimeOffset.UtcNow.AddMinutes(30));
+            }, IsAsync);
+
+            var policy = new BearerTokenAuthenticationPolicy(credential, "scope");
+            MockTransport transport = CreateMockTransport(Enumerable.Range(0, 100).Select(i => new MockResponse(200)).ToArray());
```

Comment:
You can use the constructor that takes factory func here.

Reply:
Fixed